### PR TITLE
Dev singularity

### PR DESCRIPTION
**Pull request by F-said**
- created docker image file
- created singularity recipe

### CODE REVIEW

File: container/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Instruments Container 
+
+## Overview
+This folder contains a Docker file and a Singularity recipe file that can both be used to build containers using their respective software. 
+
+The singularity image can only be built on Linux with a user that has root access. Follow [this tutorial](https://ndclab.github.io/wiki/docs/technical-docs/docker-usage.html) to build a Singularity image on a non-Linux OS.
```
Comment (georgebuzzell) [inline container/README.md:6]: does this not pose an issue for us on the hpc, given that we do not have root? @F-said

File: container/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+Additionally, the singularity image is intended to be used on the HPC cluster, whereas the docker image can be used on any local operating system.
+
+
+## Background
+Setting	up software by hand is no trivial task.	
+
+Especially in the field of open source software, the success of disseminating a project is dependent on how reliably one can set up and run a project on    
+their own machine, whether that	be a local PC or a sophisticated computing cluster.
+
+Containerization tools such as Docker & Singularity provide the set of technologies necessary to quickly replicate system tools, libraries, data, environment variables, and more. This ensures that software is reproducible and maintainable for the lifetime of a project, independent of how an operating system changes over time.
```
Comment (georgebuzzell) [inline container/README.md:17]: should we add a caveat that the kernel itself is not containerized, tho? i.e. singularity/docker are great, and solve a lot of issues, but are not perfect. @F-said

File: container/singularity.recipe [hunk 3]
```diff
@@ -0,0 +1,62 @@
+Bootstrap: docker
+From: continuumio/miniconda3
+
+%help
+
+	.__   __.  _______   ______  __          ___      .______   
+	|  \ |  | |       \ /      ||  |        /   \     |   _  \
```
Comment (georgebuzzell) [inline container/singularity.recipe:7]: ummmm, LOVE THIS!! :) @F-said

File: container/singularity.recipe [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+	apt-get -y upgrade bash-completion \
+	 build-essential \
+	 curl \
+	 git git-core \
+	 golang \
+	 htop \
+	 locales \
+	 man \
+	 nmap \
+	 python3-pip \
```
Comment (georgebuzzell) [inline container/singularity.recipe:46]: ditto to comment above, pip included in conda/miniconda, no? @F-said

**Review by georgebuzzell — CHANGES_REQUESTED** [review]
@F-said some minor comments/questions

File: container/singularity.recipe (hunk 4 above)
Comment (SDOsmany) [reply container/singularity.recipe:46]: @georgebuzzell I think miniconda/conda  comes with pip but he is installing pip3 which is a newer version. I think the conda pip version is pip2
@F-said is this correct?

**Comment by F-said** [comment]
Comments resolved, ready for merge

**Pull request merged**